### ISSUE TITLE
Error in Quadrotor runMixedIntegerForest example

### ISSUE THREAD

**Issue by JanaMHartman** [RobotLocomotion/drake#1792]
In the runMixedIntegerForest example, the script fails with a subscripted assignment dimension mismatch, and also produces a Mosek max time error.  The example runMixedIntegerSimpleForest example runs and produces a trajectory that can be run in the visuliazer, although there is a Mosek stall error in the output.

runMixedIntegerForest Output:
attempting to launch the drake director
Running mixed-integer convex program for 3rd-degree polynomial...

MOSEK Version 7.1.0.45 (Build date: 2016-1-25 06:52:46)
Copyright (c) 1998-2016 MOSEK ApS, Denmark. WWW: http://mosek.com
Platform: Linux/64-X86

Computer
  Platform               : Linux/64-X86    

Problem
  Name                   :                 
  Objective sense        : min             
  Type                   : CONIC (conic optimization problem)
  Constraints            : 7200            
  Cones                  : 211             
  Scalar variables       : 1424            
  Matrix variables       : 0               
  Integer variables      : 56              

Optimizer started.
MOSEK warning 500: The license expires in 3 days.
MOSEK warning 500: The license expires in 3 days.
Mixed integer optimizer started.
Optimizer  - threads                : 6
BRANCHES RELAXS   ACT_NDS  BEST_INT_OBJ         BEST_RELAX_OBJ       REL_GAP(%)  TIME  
0        1        0        NA                   6.3452378842e-02     NA          0.5   
0        1        1        NA                   6.3452378842e-02     NA          0.6   
0        1        1        NA                   6.3452378842e-02     NA          1.1   
0        1        1        NA                   6.3452378842e-02     NA          1.7   
0        1        1        NA                   6.3452378842e-02     NA          2.3   
0        1        1        NA                   6.3452378842e-02     NA          2.9   
1        2        2        NA                   6.3452378842e-02     NA          3.4   
3        4        4        NA                   6.3452390502e-02     NA          4.1   
7        8        4        NA                   6.3452390844e-02     NA          5.0   
9        10       6        NA                   6.3452390844e-02     NA          5.6   
12       13       7        NA                   6.3452392102e-02     NA          6.4   
16       17       7        NA                   6.3452393473e-02     NA          7.5   
20       21       11       NA                   6.3452393473e-02     NA          8.4   
24       25       13       NA                   6.3452393473e-02     NA          9.4   
33       34       16       NA                   6.3452423195e-02     NA          11.0  
45       46       26       NA                   6.3452428728e-02     NA          13.5  
69       71       40       NA                   6.3452432197e-02     NA          17.7  
105      107      48       NA                   6.3452432197e-02     NA          22.5  
153      155      58       NA                   6.3452432197e-02     NA          25.3  
207      210      62       NA                   6.3452432197e-02     NA          33.3  
267      272      96       NA                   6.3452432197e-02     NA          45.8  
363      374      148      NA                   6.3452449019e-02     NA          61.1  
494      521      173      NA                   6.3452471269e-02     NA          80.2  
646      675      201      NA                   6.3452471269e-02     NA          91.4  
796      836      231      NA                   6.3452483138e-02     NA          102.2 
952      998      233      NA                   6.3452593400e-02     NA          119.6 
1110     1158     269      NA                   6.3452614495e-02     NA          132.8 
1276     1324     303      NA                   6.3452614495e-02     NA          139.2 
1440     1489     331      NA                   6.3452614495e-02     NA          146.3 
1606     1655     339      NA                   6.9689476733e-02     NA          151.1 
1771     1820     350      NA                   8.0828355571e-02     NA          158.6 
1927     1983     350      NA                   8.1084790912e-02     NA          171.7 
2080     2142     359      NA                   8.1085120043e-02     NA          185.6 
2231     2297     382      NA                   8.1119886060e-02     NA          196.3 
2391     2460     396      NA                   1.5120289582e-01     NA          205.1 
2544     2621     415      NA                   3.3154419962e-01     NA          217.1 
2703     2782     448      NA                   3.3163434607e-01     NA          225.6 
2867     2947     454      NA                   3.3216842487e-01     NA          233.4 
3034     3114     451      NA                   3.5860460766e-01     NA          237.6 
3190     3275     421      NA                   5.4687316916e-01     NA          252.1 
3353     3438     412      NA                   5.4687379024e-01     NA          260.8 
3518     3603     393      NA                   5.4687379024e-01     NA          268.0 
3683     3768     394      NA                   5.4687379024e-01     NA          274.1 
3840     3926     377      NA                   5.6771386662e-01     NA          285.3 
3994     4086     351      NA                   8.2725359417e-01     NA          295.9 
4151     4248     328      NA                   9.4885046662e-01     NA          306.1 
4319     4416     312      NA                   1.0958959846e+00     NA          312.8 
4482     4580     287      NA                   1.3964208837e+00     NA          324.5 
4639     4738     264      NA                   2.8314054818e+00     NA          343.2 
4787     4889     288      NA                   2.8314054818e+00     NA          361.4 
4937     5044     314      NA                   2.8314054818e+00     NA          373.8 
5081     5193     304      NA                   2.8314131351e+00     NA          394.2 
5223     5340     308      NA                   2.8314131351e+00     NA          412.3 
5371     5492     308      NA                   3.0099940325e+00     NA          430.1 
5529     5651     316      NA                   3.0099943833e+00     NA          442.9 
5684     5809     313      NA                   3.0467821795e+00     NA          453.9 
5839     5965     300      NA                   3.0467824390e+00     NA          461.3 
6000     6127     287      NA                   3.1968049474e+00     NA          470.5 
6164     6292     265      NA                   3.3554524597e+00     NA          477.3 
6323     6455     236      NA                   3.6456747538e+00     NA          485.2 
6490     6622     205      NA                   3.7525376854e+00     NA          490.5 
6638     6779     173      NA                   4.0894353988e+00     NA          502.6 
6799     6940     152      NA                   6.5875332149e+00     NA          507.8 
6949     7090     156      NA                   8.0045275810e+00     NA          512.6 
7105     7246     126      NA                   9.9865004553e+00     NA          515.5 
7230     7373     133      NA                   9.9963568883e+00     NA          527.8 
7359     7510     146      NA                   1.0555638126e+01     NA          540.9 
7498     7652     159      NA                   1.0555639269e+01     NA          550.0 
7644     7798     179      NA                   1.8469544646e+01     NA          559.6 
7797     7951     182      NA                   1.8652661576e+01     NA          564.8 
7949     8104     176      NA                   1.8652856697e+01     NA          568.9 
8100     8256     171      NA                   2.1894717981e+01     NA          575.4 
8247     8405     166      NA                   2.1894717981e+01     NA          581.2 
8394     8555     133      NA                   2.1896722430e+01     NA          590.9 

Objective of best integer solution : Not available.
Best objective bound               : 2.189672242956e+01      
Construct solution objective       : Not employed
Construct solution # roundings     : 0
User objective cut value           : 0
Number of cuts generated           : 0
Number of branches                 : 8394
Number of relaxations solved       : 8555
Number of interior point iterations: 151497
Number of simplex iterations       : 0
Time spend presolving the root     : 0.04
Time spend in the heuristic        : 0.00
Time spend in the sub optimizers   : 0.00
  Time spend optimizing the root   : 0.49
Mixed integer optimizer terminated. Time: 599.50

Optimizer terminated. Time: 599.51  

Integer solution solution summary
  Problem status  : UNKNOWN
  Solution status : UNKNOWN
  Primal.  obj: 0.0000000000e+00    Viol.  con: 1e+01    var: 0e+00    cones: 0e+00    itg: 0e+00  
Optimizer summary
  Optimizer                 -                        time: 599.51  
    Interior-point          - iterations : 0         time: 0.00    
      Basis identification  -                        time: 0.00    
        Primal              - iterations : 0         time: 0.00    
        Dual                - iterations : 0         time: 0.00    
        Clean primal        - iterations : 0         time: 0.00    
        Clean dual          - iterations : 0         time: 0.00    
        Clean primal-dual   - iterations : 0         time: 0.00    
    Simplex                 -                        time: 0.00    
      Primal simplex        - iterations : 0         time: 0.00    
      Dual simplex          - iterations : 0         time: 0.00    
      Primal-dual simplex   - iterations : 0         time: 0.00    
    Mixed integer           - relaxations: 8555      time: 599.50  

Mosek error: MSK_RES_TRM_MAX_TIME ()
Elapsed time is 601.803632 seconds.
done!
Running semidefinite program for 5th-degree polynomial...

MOSEK Version 7.1.0.45 (Build date: 2016-1-25 06:52:46)
Copyright (c) 1998-2016 MOSEK ApS, Denmark. WWW: http://mosek.com
Platform: Linux/64-X86

Computer
  Platform               : Linux/64-X86    

Problem
  Name                   :                 
  Objective sense        : min             
  Type                   : CONIC (conic optimization problem)
  Constraints            : 2437            
  Cones                  : 1               
  Scalar variables       : 1202            
  Matrix variables       : 210             
  Integer variables      : 0               

Optimizer started.
MOSEK warning 500: The license expires in 3 days.
MOSEK warning 500: The license expires in 3 days.
Conic interior-point optimizer started.
Presolve started.
Linear dependency checker started.
Linear dependency checker terminated.
Eliminator started.
Total number of eliminations : 81
Eliminator terminated.
Eliminator - tries                  : 1                 time                   : 0.00            
Eliminator - elim's                 : 81              
Lin. dep.  - tries                  : 1                 time                   : 0.00            
Lin. dep.  - number                 : 0               
Presolve terminated. Time: 0.00    
Optimizer  - threads                : 6               
Optimizer  - solved problem         : the primal      
Optimizer  - Constraints            : 1285
Optimizer  - Cones                  : 1
Optimizer  - Scalar variables       : 51                conic                  : 44              
Optimizer  - Semi-definite variables: 210               scalarized             : 1260            
Factor     - setup time             : 0.00              dense det. time        : 0.00            
Factor     - ML order time          : 0.00              GP order time          : 0.00            
Factor     - nonzeros before factor : 4735              after factor           : 4735            
Factor     - dense dim.             : 0                 flops                  : 7.17e+04        
ITE PFEAS    DFEAS    GFEAS    PRSTATUS   POBJ              DOBJ              MU       TIME  
0   1.0e+00  7.3e+00  2.0e+00  0.00e+00   1.000000000e+00   0.000000000e+00   1.0e+00  0.01  
1   7.2e-02  5.3e-01  1.4e-01  3.53e-01   4.298668581e-02   -5.123374599e-02  7.2e-02  0.02  
2   4.0e-03  2.9e-02  8.0e-03  9.56e-01   -4.353120488e-02  -4.881797981e-02  4.0e-03  0.04  
3   2.1e-04  1.6e-03  4.2e-04  9.97e-01   -4.870400427e-02  -4.898447089e-02  2.1e-04  0.05  
4   1.1e-05  8.1e-05  2.2e-05  1.00e+00   -4.898705132e-02  -4.900169669e-02  1.1e-05  0.06  
5   5.7e-07  4.2e-06  1.1e-06  1.00e+00   -4.900213763e-02  -4.900289503e-02  5.7e-07  0.08  
6   3.7e-09  1.5e-08  4.0e-09  1.00e+00   -4.900296878e-02  -4.900297140e-02  2.0e-09  0.09  
Interior-point optimizer terminated. Time: 0.09. 

Optimizer terminated. Time: 0.09    

Interior-point solution summary
  Problem status  : PRIMAL_AND_DUAL_FEASIBLE
  Solution status : OPTIMAL
  Primal.  obj: -4.9002968777e-02   Viol.  con: 5e-09    var: 0e+00    barvar: 0e+00    cones: 0e+00  
  Dual.    obj: -4.9002971401e-02   Viol.  con: 0e+00    var: 2e-08    barvar: 3e-09    cones: 0e+00  
Optimizer summary
  Optimizer                 -                        time: 0.09    
    Interior-point          - iterations : 6         time: 0.09    
      Basis identification  -                        time: 0.00    
        Primal              - iterations : 0         time: 0.00    
        Dual                - iterations : 0         time: 0.00    
        Clean primal        - iterations : 0         time: 0.00    
        Clean dual          - iterations : 0         time: 0.00    
        Clean primal-dual   - iterations : 0         time: 0.00    
    Simplex                 -                        time: 0.00    
      Primal simplex        - iterations : 0         time: 0.00    
      Dual simplex          - iterations : 0         time: 0.00    
      Primal-dual simplex   - iterations : 0         time: 0.00    
    Mixed integer           - relaxations: 0         time: 0.00    

Elapsed time is 0.344190 seconds.
Subscripted assignment dimension mismatch.

Error in MISOSTrajectoryProblem/solveTrajectory (line 279)
          coeffs(i,k,:) = fliplr(reshape(ct,1,[]));

Error in runMixedIntegerEnvironment (line 100)
ytraj = prob.solveTrajectory(start, goal, safe_regions, safe_region_assignments);

Error in runMixedIntegerForest (line 28)
runMixedIntegerEnvironment(r, start, goal, lb, ub, seeds, degree, n_segments, n_regions);

**Comment by rdeits** [comment]
Yeah, it looks like Mosek just ran out of time (which may be a function of your CPU speed or system load) and produced a garbage solution. Can you try increasing the timeout for the Mosek solve, just to make sure that's all that's going on? The line to change is line 254 of drake/examples/Quadrotor/MISOSTrajectoryProblem.m:

```matlab
diagnostics = optimize(constraints, objective, sdpsettings('solver', 'mosek', 'mosek.MSK_DPAR_MIO_TOL_REL_GAP', 1e-2, 'mosek.MSK_DPAR_MIO_MAX_TIME', 600, 'verbose', 3));
```

**Comment by JanaMHartman** [comment]
I changed the time from 600 to 1200.  It now tells me the problem is Primal Infeasible.  The objective function value jumps abruptly in the initial optimization.  I've included the beginning and the end when the jump occurs.  The final error, after the 5th degree run, is the same as before since a solution is still not found.


runMixedIntegerForest
Running mixed-integer convex program for 3rd-degree polynomial...

MOSEK Version 7.1.0.45 (Build date: 2016-1-25 06:52:46)
Copyright (c) 1998-2016 MOSEK ApS, Denmark. WWW: http://mosek.com
Platform: Linux/64-X86

Computer
  Platform               : Linux/64-X86    

Problem
  Name                   :                 
  Objective sense        : min             
  Type                   : CONIC (conic optimization problem)
  Constraints            : 7200            
  Cones                  : 211             
  Scalar variables       : 1424            
  Matrix variables       : 0               
  Integer variables      : 56              

Optimizer started.
MOSEK warning 500: The license expires in 1 days.
MOSEK warning 500: The license expires in 1 days.
Mixed integer optimizer started.
Optimizer  - threads                : 6
BRANCHES RELAXS   ACT_NDS  BEST_INT_OBJ         BEST_RELAX_OBJ       REL_GAP(%)  TIME  
0        1        0        NA                   6.3452378842e-02     NA          0.5   
0        1        1        NA                   6.3452378842e-02     NA          0.6   
0        1        1        NA                   6.3452378842e-02     NA          1.2   
0        1        1        NA                   6.3452378842e-02     NA          1.7   

...

9633     9832     22       NA                   6.6614503408e+01     NA          699.9 
9649     9848     16       NA                   6.6614503408e+01     NA          701.3 
9661     9861     12       NA                   6.6614543163e+01     NA          703.4 
9669     9869     12       NA                   6.6614546845e+01     NA          704.5 
9679     9879     6        NA                   1.0000000000e+300    NA          706.0 
9683     9883     8        NA                   1.0000000000e+300    NA          706.6 
9688     9888     9        NA                   1.0000000000e+300    NA          707.5 
9691     9891     10       NA                   1.0000000000e+300    NA          708.6 
9695     9896     8        NA                   1.0000000000e+300    NA          710.7 
9699     9901     10       NA                   1.0000000000e+300    NA          712.4 
9704     9907     11       NA                   1.0000000000e+300    NA          714.2 
9708     9911     13       NA                   1.0000000000e+300    NA          714.7 
9717     9921     10       NA                   1.0000000000e+300    NA          716.8 
9722     9926     11       NA                   1.0000000000e+300    NA          717.4 
9726     9930     13       NA                   1.0000000000e+300    NA          717.8 
9735     9939     4        NA                   1.0000000000e+300    NA          718.6 
9737     9941     4        NA                   1.0000000000e+300    NA          718.9 
9739     9943     4        NA                   1.0000000000e+300    NA          719.2 
9741     9945     2        NA                   1.0000000000e+300    NA          719.4 
9742     9946     3        NA                   1.0000000000e+300    NA          719.9 
9745     9949     4        NA                   1.0000000000e+300    NA          720.3 
9748     9952     5        NA                   1.0000000000e+300    NA          720.6 
9751     9955     6        NA                   1.0000000000e+300    NA          720.9 
9754     9958     5        NA                   1.0000000000e+300    NA          721.2 
9758     9962     3        NA                   1.0000000000e+300    NA          721.6 
9761     9965     2        NA                   1.0000000000e+300    NA          721.8 
9763     9967     0        NA                   1.0000000000e+300    NA          722.1 

Objective of best integer solution : Not available.
Best objective bound               : 1.000000000000e+300     
Construct solution objective       : Not employed
Construct solution # roundings     : 0
User objective cut value           : 0
Number of cuts generated           : 0
Number of branches                 : 9763
Number of relaxations solved       : 9967
Number of interior point iterations: 179659
Number of simplex iterations       : 0
Time spend presolving the root     : 0.04
Time spend in the heuristic        : 0.00
Time spend in the sub optimizers   : 0.00
  Time spend optimizing the root   : 0.49
Mixed integer optimizer terminated. Time: 722.09

Optimizer terminated. Time: 722.11  

Integer solution solution summary
  Problem status  : PRIMAL_INFEASIBLE
  Solution status : UNKNOWN
  Primal.  obj: 0.0000000000e+00    Viol.  con: 1e+01    var: 0e+00    cones: 0e+00    itg: 0e+00  
Optimizer summary
  Optimizer                 -                        time: 722.11  
    Interior-point          - iterations : 0         time: 0.00    
      Basis identification  -                        time: 0.00    
        Primal              - iterations : 0         time: 0.00    
        Dual                - iterations : 0         time: 0.00    
        Clean primal        - iterations : 0         time: 0.00    
        Clean dual          - iterations : 0         time: 0.00    
        Clean primal-dual   - iterations : 0         time: 0.00    
    Simplex                 -                        time: 0.00    
      Primal simplex        - iterations : 0         time: 0.00    
      Dual simplex          - iterations : 0         time: 0.00    
      Primal-dual simplex   - iterations : 0         time: 0.00    
    Mixed integer           - relaxations: 9967      time: 722.09  

Elapsed time is 724.299104 seconds.
done!
